libqtile/xcbq.py

```diff
@@ -316,20 +317,20 @@ def alloc_color(self, color):
         """
             Flexible color allocation.
         """
-        if color.startswith("#"):
-            if len(color) != 7:
+        try:
+            return self.conn.conn.core.AllocNamedColor(
+                self.cid, len(color), color
+            ).reply()
+        except xcffib.xproto.NameError:
+            if len(color) < 6:
                 raise ValueError("Invalid color: %s" % color)
```

Comment:
Yep, maybe `except xcffib.xproto.NameError:` could just `pass`, and its block could be enclosed in another `try` that catches IndexError and other possible exceptions caused by bad `color` strings, for example containing non-hexadecimal characters (which would break the `int` calls I guess), but this is all fussy optimization after all, I agree ;)